llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -3216,13 +3216,16 @@ const SCEV *ScalarEvolution::getMulExpr(SmallVectorImpl<const SCEV *> &Ops,
         };
       }
 
-      // Try to fold (C * D /u C) -> D, if C is a power-of-2 and D is a multiple
-      //  of C.
+      // Try to fold (C1 * D /u C2) -> C1/C2 * D, if C1 and C2 are powers-of-2,
+      // D is a multiple of C2, and C1 is a multiple of C1.
```

Comment:
Should probably say C1 is a multiple of C2?

Reply:
Yep, I adjusted this in a follow-up PR (https://github.com/llvm/llvm-project/pull/157555), but can also fix separately.